### PR TITLE
Add test case for AopTargetUtils

### LINKED ISSUE #1212: Add test case for AopTargetUtils

### PR DESCRIPTION
**Pull request by zhaoyuguang**
Fixes #1212

### CODE REVIEW

File: elasticjob-lite/elasticjob-lite-spring/elasticjob-lite-spring-namespace/src/test/java/org/apache/shardingsphere/elasticjob/lite/spring/namespace/job/util/AopTargetUtilsTest.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+import org.apache.shardingsphere.elasticjob.api.ElasticJob;
+import org.junit.Test;
+import org.springframework.aop.framework.ProxyFactory;
+import org.springframework.aop.support.AopUtils;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class AopTargetUtilsTest {
```
Comment (dongzl) [inline elasticjob-lite/elasticjob-lite-spring/elasticjob-lite-spring-namespace/src/test/java/org/apache/shardingsphere/elasticjob/lite/spring/namespace/job/util/AopTargetUtilsTest.java]: should be `final` class.

File: elasticjob-lite/elasticjob-lite-spring/elasticjob-lite-spring-namespace/src/test/java/org/apache/shardingsphere/elasticjob/lite/spring/namespace/job/util/TargetJob.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+import org.apache.shardingsphere.elasticjob.api.ElasticJob;
+import org.apache.shardingsphere.elasticjob.api.ShardingContext;
+
+public class TargetJob implements ElasticJob {
+
+    public void execute(final ShardingContext shardingContext) {
+
+    }
+}
```
Comment (dongzl) [inline elasticjob-lite/elasticjob-lite-spring/elasticjob-lite-spring-namespace/src/test/java/org/apache/shardingsphere/elasticjob/lite/spring/namespace/job/util/TargetJob.java]: add blank line at end of file.

File: elasticjob-lite/elasticjob-lite-spring/elasticjob-lite-spring-namespace/src/test/java/org/apache/shardingsphere/elasticjob/lite/spring/namespace/job/util/AopTargetUtilsTest.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        AopTargetUtils.getTarget(proxy);
+    }
+
+    @Test
+    public void noneProxyForGetTarget() {
+        ElasticJob proxy = new TargetJob();
+        assertFalse(AopUtils.isAopProxy(proxy));
+        AopTargetUtils.getTarget(proxy);
+    }
+}
```
Comment (dongzl) [inline elasticjob-lite/elasticjob-lite-spring/elasticjob-lite-spring-namespace/src/test/java/org/apache/shardingsphere/elasticjob/lite/spring/namespace/job/util/AopTargetUtilsTest.java]: add blank line at end of file.

File: elasticjob-lite/elasticjob-lite-spring/elasticjob-lite-spring-namespace/src/test/java/org/apache/shardingsphere/elasticjob/lite/spring/namespace/job/util/AopTargetUtilsTest.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+import org.springframework.aop.framework.ProxyFactory;
+import org.springframework.aop.support.AopUtils;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class AopTargetUtilsTest {
+
+    @Test
+    public void jdkDynamicProxyForGetTarget() {
```
Comment (dongzl) [inline elasticjob-lite/elasticjob-lite-spring/elasticjob-lite-spring-namespace/src/test/java/org/apache/shardingsphere/elasticjob/lite/spring/namespace/job/util/AopTargetUtilsTest.java]: unit test method suggest start with `assert`.

File: elasticjob-lite/elasticjob-lite-spring/elasticjob-lite-spring-namespace/src/test/java/org/apache/shardingsphere/elasticjob/lite/spring/namespace/job/util/AopTargetUtilsTest.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        ProxyFactory pf = new ProxyFactory(new TargetJob());
+        pf.addInterface(ElasticJob.class);
+        ElasticJob proxy = (ElasticJob) pf.getProxy();
+        assertTrue(AopUtils.isJdkDynamicProxy(proxy));
+        AopTargetUtils.getTarget(proxy);
+    }
+
+    @Test
+    public void cglibProxyForGetTarget() {
```
Comment (dongzl) [inline elasticjob-lite/elasticjob-lite-spring/elasticjob-lite-spring-namespace/src/test/java/org/apache/shardingsphere/elasticjob/lite/spring/namespace/job/util/AopTargetUtilsTest.java]: unit test method suggest start with `assert`.

File: elasticjob-lite/elasticjob-lite-spring/elasticjob-lite-spring-namespace/src/test/java/org/apache/shardingsphere/elasticjob/lite/spring/namespace/job/util/AopTargetUtilsTest.java [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+        ProxyFactory pf = new ProxyFactory(new TargetJob());
+        pf.setProxyTargetClass(true);
+        ElasticJob proxy = (ElasticJob) pf.getProxy();
+        assertTrue(AopUtils.isCglibProxy(proxy));
+        AopTargetUtils.getTarget(proxy);
+    }
+
+    @Test
+    public void noneProxyForGetTarget() {
```
Comment (dongzl) [inline elasticjob-lite/elasticjob-lite-spring/elasticjob-lite-spring-namespace/src/test/java/org/apache/shardingsphere/elasticjob/lite/spring/namespace/job/util/AopTargetUtilsTest.java]: unit test method suggest start with `assert`.

File: elasticjob-lite/elasticjob-lite-spring/elasticjob-lite-spring-namespace/src/test/java/org/apache/shardingsphere/elasticjob/lite/spring/namespace/job/util/AopTargetUtilsTest.java [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+import org.apache.shardingsphere.elasticjob.api.ElasticJob;
+import org.junit.Test;
+import org.springframework.aop.framework.ProxyFactory;
+import org.springframework.aop.support.AopUtils;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class AopTargetUtilsTest {
+
```
Comment (dongzl) [inline elasticjob-lite/elasticjob-lite-spring/elasticjob-lite-spring-namespace/src/test/java/org/apache/shardingsphere/elasticjob/lite/spring/namespace/job/util/AopTargetUtilsTest.java]: add four white space, keep indents consistent with the previous one.

File: elasticjob-lite/elasticjob-lite-spring/elasticjob-lite-spring-namespace/src/test/java/org/apache/shardingsphere/elasticjob/lite/spring/namespace/job/util/AopTargetUtilsTest.java [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+    @Test
+    public void jdkDynamicProxyForGetTarget() {
+        ProxyFactory pf = new ProxyFactory(new TargetJob());
+        pf.addInterface(ElasticJob.class);
+        ElasticJob proxy = (ElasticJob) pf.getProxy();
+        assertTrue(AopUtils.isJdkDynamicProxy(proxy));
+        AopTargetUtils.getTarget(proxy);
+    }
+
```
Comment (dongzl) [inline elasticjob-lite/elasticjob-lite-spring/elasticjob-lite-spring-namespace/src/test/java/org/apache/shardingsphere/elasticjob/lite/spring/namespace/job/util/AopTargetUtilsTest.java]: add four white space, keep indents consistent with the previous one.

File: elasticjob-lite/elasticjob-lite-spring/elasticjob-lite-spring-namespace/src/test/java/org/apache/shardingsphere/elasticjob/lite/spring/namespace/job/util/AopTargetUtilsTest.java [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+    @Test
+    public void cglibProxyForGetTarget() {
+        ProxyFactory pf = new ProxyFactory(new TargetJob());
+        pf.setProxyTargetClass(true);
+        ElasticJob proxy = (ElasticJob) pf.getProxy();
+        assertTrue(AopUtils.isCglibProxy(proxy));
+        AopTargetUtils.getTarget(proxy);
+    }
+
```
Comment (dongzl) [inline elasticjob-lite/elasticjob-lite-spring/elasticjob-lite-spring-namespace/src/test/java/org/apache/shardingsphere/elasticjob/lite/spring/namespace/job/util/AopTargetUtilsTest.java]: add four white space, keep indents consistent with the previous one.

File: elasticjob-lite/elasticjob-lite-spring/elasticjob-lite-spring-namespace/src/test/java/org/apache/shardingsphere/elasticjob/lite/spring/namespace/job/util/TargetJob.java [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.elasticjob.lite.spring.namespace.job.util;
+
+import org.apache.shardingsphere.elasticjob.api.ElasticJob;
+import org.apache.shardingsphere.elasticjob.api.ShardingContext;
+
+public class TargetJob implements ElasticJob {
+
```
Comment (dongzl) [inline elasticjob-lite/elasticjob-lite-spring/elasticjob-lite-spring-namespace/src/test/java/org/apache/shardingsphere/elasticjob/lite/spring/namespace/job/util/TargetJob.java]: add four white space, keep indents consistent with the previous one.

File: elasticjob-lite/elasticjob-lite-spring/elasticjob-lite-spring-namespace/src/test/java/org/apache/shardingsphere/elasticjob/lite/spring/namespace/job/util/TargetJob.java [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.elasticjob.lite.spring.namespace.job.util;
+
+import org.apache.shardingsphere.elasticjob.api.ElasticJob;
+import org.apache.shardingsphere.elasticjob.api.ShardingContext;
+
+public class TargetJob implements ElasticJob {
```
Comment (dongzl) [inline elasticjob-lite/elasticjob-lite-spring/elasticjob-lite-spring-namespace/src/test/java/org/apache/shardingsphere/elasticjob/lite/spring/namespace/job/util/TargetJob.java:23]: should be `final` class.

**Review by dongzl — CHANGES_REQUESTED** [review]

**Comment by zhaoyuguang** [comment]
Thanks for guide,  I will fixed as soon

File: elasticjob-lite/elasticjob-lite-spring/elasticjob-lite-spring-namespace/src/test/java/org/apache/shardingsphere/elasticjob/lite/spring/namespace/job/util/TargetJob.java (hunk 11 above)
Comment (zhaoyuguang) [reply elasticjob-lite/elasticjob-lite-spring/elasticjob-lite-spring-namespace/src/test/java/org/apache/shardingsphere/elasticjob/lite/spring/namespace/job/util/TargetJob.java:23]: CGLIB testcase use it，so this not add

File: elasticjob-lite/elasticjob-lite-spring/elasticjob-lite-spring-namespace/src/test/java/org/apache/shardingsphere/elasticjob/lite/spring/namespace/job/util/AopTargetUtilsTest.java [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+public final class AopTargetUtilsTest {
+    
+    @Test
+    public void assertJdkDynamicProxyForGetTarget() {
+        ElasticJob target = new TargetJob();
+        ProxyFactory pf = new ProxyFactory(target);
+        pf.addInterface(ElasticJob.class);
+        ElasticJob proxy = (ElasticJob) pf.getProxy();
+        assertTrue(AopUtils.isJdkDynamicProxy(proxy));
+        assertEquals(AopTargetUtils.getTarget(proxy), target);
```
Comment (dongzl) [inline elasticjob-lite/elasticjob-lite-spring/elasticjob-lite-spring-namespace/src/test/java/org/apache/shardingsphere/elasticjob/lite/spring/namespace/job/util/AopTargetUtilsTest.java]: For multiple parameter asserts, `assertThat` should be used.

File: elasticjob-lite/elasticjob-lite-spring/elasticjob-lite-spring-namespace/src/test/java/org/apache/shardingsphere/elasticjob/lite/spring/namespace/job/util/AopTargetUtilsTest.java [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+    }
+    
+    @Test
+    public void assertCglibProxyForGetTarget() {
+        ElasticJob target = new TargetJob();
+        ProxyFactory pf = new ProxyFactory(target);
+        pf.setProxyTargetClass(true);
+        ElasticJob proxy = (ElasticJob) pf.getProxy();
+        assertTrue(AopUtils.isCglibProxy(proxy));
+        assertEquals(AopTargetUtils.getTarget(proxy), target);
```
Comment (dongzl) [inline elasticjob-lite/elasticjob-lite-spring/elasticjob-lite-spring-namespace/src/test/java/org/apache/shardingsphere/elasticjob/lite/spring/namespace/job/util/AopTargetUtilsTest.java]: For multiple parameter asserts, `assertThat` should be used.

File: elasticjob-lite/elasticjob-lite-spring/elasticjob-lite-spring-namespace/src/test/java/org/apache/shardingsphere/elasticjob/lite/spring/namespace/job/util/AopTargetUtilsTest.java [hunk 14]
```diff
@@ -0,0 +1,57 @@
...
+    }
+    
+    @Test
+    public void assertNoneProxyForGetTarget() {
+        ElasticJob proxy = new TargetJob();
+        assertFalse(AopUtils.isAopProxy(proxy));
+        assertEquals(AopTargetUtils.getTarget(proxy), proxy);
```
Comment (dongzl) [inline elasticjob-lite/elasticjob-lite-spring/elasticjob-lite-spring-namespace/src/test/java/org/apache/shardingsphere/elasticjob/lite/spring/namespace/job/util/AopTargetUtilsTest.java]: For multiple parameter asserts, `assertThat` should be used.

File: elasticjob-lite/elasticjob-lite-spring/elasticjob-lite-spring-namespace/src/test/java/org/apache/shardingsphere/elasticjob/lite/spring/namespace/job/util/TargetJob.java (hunk 11 above)
Comment (dongzl) [reply elasticjob-lite/elasticjob-lite-spring/elasticjob-lite-spring-namespace/src/test/java/org/apache/shardingsphere/elasticjob/lite/spring/namespace/job/util/TargetJob.java:23]: OK

**Review by dongzl — CHANGES_REQUESTED** [review]

**Review by terrymanu — APPROVED** [review]

**Pull request merged**